service/controller/app/v1/controllercontext/context.go

```diff
@@ -0,0 +1,29 @@
+package controllercontext
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+)
+
+type contextKey string
+
+const controllerKey contextKey = "controller"
+
+type Context struct {
+	G8sClient versioned.Interface
+}
+
+func NewContext(ctx context.Context, c Context) context.Context {
```

Comment:
IMO when `FromContext` returns a pointer, `NewContext` should take one.

Reply:
we could do it, but I don't want to take apart from its original source, such as `aws-operator`